### PR TITLE
Add weekly tests

### PR DESCRIPTION
**Pull request by Ericgig**
Runs tests automatically each week.
Any failure should trigger an email to those watching this repo.

### CODE REVIEW

File: .github/workflows/tests.yml
Comment (Ericgig) [inline .github/workflows/tests.yml:14]:
```suggestion
    [push, pull_request]
```

**Comment by Ericgig** [comment]
@AGaliciaMartinez @quantshah,
Could you review this.
For security reason, new workflow don't have access to secret during PR so it needs to be merged to test the sending email step. 

If it work as intended, each week tests would run and send and email to the google group on failure.
We can trigger the tests manually and send email to a desired destination. This is mostly to allow me to debug potential issues with the email server without spamming the google group.

**Review by AGaliciaMartinez — APPROVED** [review]
Thank for the merge @Ericgig. Seems good to me, lets merge it and test if ti works. Perhaps I have one question about this PR. Will the emails be sent by default to anyone in qutip-admin? if so, I would suggest to somehow restrict this as it could be too much spam for some that are not directly involved in writing code.

**Comment by Ericgig** [comment]
Thank you for reviewing. I will wait until January to merge.

Yes this will send email to the qutip-admin email list when failing. 
We should not use hard coded personal email here. 
Ideally a qutip-dev email would be better.

**Review by hodgestar — APPROVED** [review]

**Pull request merged**
